clusterloader2/pkg/measurement/util/runtimeobjects/runtimeobjects.go

```diff
@@ -302,20 +302,15 @@ func GetReplicasFromRuntimeObject(c clientset.Interface, obj runtime.Object) (in
 	}
 }
 
-// getNumSchedulableNodesMatchingSelector returns the number of schedulable nodes matching the provided selector.
-func getNumSchedulableNodesMatchingSelector(c clientset.Interface, nodeSelector map[string]string) (int32, error) {
-	selector, err := metav1.LabelSelectorAsSelector(metav1.SetAsLabelSelector(nodeSelector))
-	if err != nil {
-		return 0, err
-	}
-	listOpts := metav1.ListOptions{LabelSelector: selector.String()}
-	list, err := client.ListNodesWithOptions(c, listOpts)
+// getNumSchedulableNodesMatchingSelector returns the number of schedulable nodes matching the provided pod spec.
+func getNumSchedulableNodesMatchingSelector(c clientset.Interface, podSpec corev1.PodSpec) (int32, error) {
+	nodes, err := client.ListNodes(c)
 	if err != nil {
 		return 0, err
 	}
 	var numSchedulableNodes int32
-	for _, node := range list {
-		if !node.Spec.Unschedulable {
+	for _, node := range nodes {
+		if podMatchesNodeLabels(podSpec, node) && !node.Spec.Unschedulable {
```

Comment:
nit: I would change the order of the conditions to utilize power of lazy evaluation :)